From a1fda68a1963a5cfb21c78c688f12e1717fc3243 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 14 May 2012 14:18:38 +0900
Subject: [PATCH 18/97] ext/dfbvideosink: pass a dummy pointer as the 3rd
 argument of TestConfiguration(), rather than NULL.

This change is a workaround and may be unnessesary for the latest DirectFB.
According to the spec, the third argument may be specified as NULL, but the
current DirectFB doesn't NULL-check the third argument of TestConfiguration()
when running in slave mode.
---
 ext/directfb/dfbvideosink.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index e52b841..2966626 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1022,6 +1022,7 @@ gst_dfbvideosink_can_blit_from_format (GstDfbVideoSink * dfbvideosink,
   DFBSurfaceDescription s_dsc;
   DFBAccelerationMask mask;
   DFBDisplayLayerConfig dlc, prev_dlc;
+  DFBDisplayLayerConfigFlags failed;
 
   g_return_val_if_fail (GST_IS_DFBVIDEOSINK (dfbvideosink), FALSE);
 
@@ -1051,7 +1052,7 @@ gst_dfbvideosink_can_blit_from_format (GstDfbVideoSink * dfbvideosink,
   dlc.pixelformat = format;
 
   ret = dfbvideosink->layer->TestConfiguration (dfbvideosink->layer, &dlc,
-      NULL);
+      &failed);
   if (ret != DFB_OK) {
     GST_DEBUG_OBJECT (dfbvideosink, "our layer refuses to operate in pixel "
         "format %s", gst_dfbvideosink_get_format_name (format));
-- 
1.7.10.4

